app/models/user.rb

```diff
@@ -13,13 +13,20 @@
 #  remember_created_at    :datetime
 #
 class User < ApplicationRecord
-  # Include default devise modules. Others available are:
-  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
   devise :database_authenticatable, :registerable,
-         :recoverable, :rememberable, :validatable
+         :recoverable, :rememberable, :validatable,
+         :omniauthable, omniauth_providers: [:google_oauth2]
 
   has_many :resources
   has_many :resource_comments
   has_many :resource_evaluations
   has_many :completed_learning_units
+
+  def self.from_omniauth(auth)
+    where(provider: auth.provider, uid: auth.uid).first_or_create do |user|
```

Comment:
Is there a reason to use first_or_create instead of find_or_create_by?

Reply:
Both could be useful, [here](https://stackoverflow.com/questions/42178674/first-or-create-vs-find-or-create-by) I leave a reference